protocol/chainlib/grpc.go

```diff
@@ -488,7 +468,7 @@ func (cp *GrpcChainProxy) SendNodeMsg(ctx context.Context, ch chan interface{},
 		}
 
 		// add the descriptor to the chainProxy cache
-		cp.descriptorsCache.setDescriptor(methodName, methodDescriptor)
+		cp.descriptorsCache.Store(methodName, methodDescriptor)
```

Comment:
prone to races, let's use loadOrStore?

Reply:
On the worst case scenario, where 2 threads are accessing the map at the same time, the same descriptor will be stored twice. I think that the current solution better than always creating the descriptor, since the outcome is the same.

Reply:
awesome, let's comment this and move on

Reply:
Fixed: 397917ff5ab5815a617a0944b4ab4f55aff40f16